### ISSUE TITLE
f2py fails on get_parameters on Linux, works on OS X

### ISSUE THREAD

**Issue by teored90** [numpy/numpy#8782]
Consider the following FORTRAN file `hello.f90`

```
SUBROUTINE qw1p1d()
  IMPLICIT NONE
  COMPLEX, PARAMETER:: ii = (0.,1.)
  WRITE(*,*) "Hello, World!"
END SUBROUTINE qw1p1d
```

This compiles fine on my OSX with Anaconda and `numpy 1.11.3`, but I get the following on a Linux machine with `numpy 1.12.0`. The problem is in the line `COMPLEX, PARAMETER:: ii = (0.,1.)`:

```
$ f2py --debug -c hello.f90 -m hello
running build
running config_cc
unifing config_cc, config, build_clib, build_ext, build commands --compiler options
running config_fc
unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options
running build_src
build_src
building extension "hello" sources
f2py options: []
f2py:> /tmp/tmpufqe22pk/src.linux-x86_64-3.5/hellomodule.c
creating /tmp/tmpufqe22pk/src.linux-x86_64-3.5
Reading fortran codes...
	Reading file 'hello.f90' (format:free)
Post-processing...
	Block: hello
			Block: qw1p1d
Traceback (most recent call last):
  File "/usr/local/bin/f2py", line 28, in <module>
    main()
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/f2py2e.py", line 648, in main
    run_compile()
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/f2py2e.py", line 633, in run_compile
    setup(ext_modules=[ext])
  File "/usr/local/lib/python3.5/dist-packages/numpy/distutils/core.py", line 169, in setup
    return old_setup(**new_attr)
  File "/usr/lib/python3.5/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.5/dist-packages/numpy/distutils/command/build.py", line 47, in run
    old_build.run(self)
  File "/usr/lib/python3.5/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/usr/lib/python3.5/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.5/dist-packages/numpy/distutils/command/build_src.py", line 148, in run
    self.build_sources()
  File "/usr/local/lib/python3.5/dist-packages/numpy/distutils/command/build_src.py", line 165, in build_sources
    self.build_extension_sources(ext)
  File "/usr/local/lib/python3.5/dist-packages/numpy/distutils/command/build_src.py", line 327, in build_extension_sources
    sources = self.f2py_sources(sources, ext)
  File "/usr/local/lib/python3.5/dist-packages/numpy/distutils/command/build_src.py", line 564, in f2py_sources
    ['-m', ext_name]+f_sources)
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/f2py2e.py", line 408, in run_main
    postlist = callcrackfortran(files, options)
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/f2py2e.py", line 329, in callcrackfortran
    postlist = crackfortran.crackfortran(files)
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 3248, in crackfortran
    postlist = postcrack(grouplist[0])
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 1885, in postcrack
    g = postcrack(g, tab=tab + '\t')
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 1904, in postcrack
    block['body'] = analyzebody(block, args, tab=tab)
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 2066, in analyzebody
    b = postcrack(b, as_, tab=tab + '\t')
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 1904, in postcrack
    block['body'] = analyzebody(block, args, tab=tab)
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 2066, in analyzebody
    b = postcrack(b, as_, tab=tab + '\t')
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 1900, in postcrack
    block['vars'] = analyzevars(block)
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 2544, in analyzevars
    params = get_parameters(vars, get_useparameters(block))
  File "/usr/local/lib/python3.5/dist-packages/numpy/f2py/crackfortran.py", line 2473, in get_parameters
    if v[0] == '(' and v[-1] == ')':
IndexError: string index out of range
 ```

**Comment by teored90** [comment]
I updated `numpy` to `1.12.0` on my OSX machine and I obtain exactly the same error as above, so this looks like a regression.

**Comment by juliantaylor** [comment]
hi,
this should be fixed in 1.12.1 which will be released soon
https://mail.scipy.org/pipermail/numpy-discussion/2017-March/076555.html
